Program.cs

```diff
@@ -1,41 +1,53 @@
 ﻿static void TreasureHunt()
```

Comment:
All methods should be after the entry point at the one of the file. In your case it means, that calling this function should be the first line and the functions can follow.